### PR TITLE
Add recipe for sdl2_mixer (needs other recipes to be merged first)

### PR DESCRIPTION
**Pull request by dschreij**
sdl2_mixer depends on sdl2, which is not yet available on conda-forge, so these builds will probably fail until this is the case.

sdl2_mixer depends on:
- ogg
- vorbis
- flac
- smpeg2
- mad

which are not yet available on conda, but for which the recipes are also included in this PR.

### CODE REVIEW

**Comment by dschreij** [comment]
The builds fail mostly because of the missing sdl2 dependency which was expected. However, there are a few other culprits:
- AppVeyor cannot unpack the flac.tar.xz package _only_ for Python 2.7 (32 and 64 bit) issuing the message `unxz is required to unarchive .xz source files.
`. The later python versions have no trouble with this.
- auto(re)conf doesn't work correctly on circleci, even though it doesn't cause any troubles on my local system (which is a Mac, admittedly). The error message is
~~~
autoreconf -fi
configure.ac:65: error: possibly undefined macro: AC_PROG_LIBTOOL
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
~~~
Anything I can do about this?

**Comment by dschreij** [comment]
@conda-forge/core, I hope if it's ok to ask you for help with this particular issue? On Windows (appveyor) the build of sdl2_mixer appears to fail before it has even started the output is 

~~~
...
Building  sdl2_mixer-2.0.1-vc9_0
Traceback (most recent call last):
  File "C:\Miniconda\Scripts\conda-build-all-script.py", line 11, in <module>
    load_entry_point('conda-build-all==1.0.0', 'console_scripts', 'conda-build-all')()
  File "C:\Miniconda\lib\site-packages\conda_build_all\cli.py", line 90, in main
    b.main()
  File "C:\Miniconda\lib\site-packages\conda_build_all\builder.py", line 263, in main
    built_dist_location = self.build(meta, build_config)
  File "C:\Miniconda\lib\site-packages\conda_build_all\builder.py", line 204, in build
    return bldpkg_path(build.build(meta.meta))
  File "C:\Miniconda\lib\site-packages\conda_build_all\build.py", line 33, in build
    shutil.rmtree(conda_build.source.WORK_DIR)
  File "C:\Miniconda\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "C:\Miniconda\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "C:\Miniconda\lib\shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "C:\Miniconda\lib\shutil.py", line 252, in rmtree
    onerror(os.remove, fullname, sys.exc_info())
  File "C:\Miniconda\lib\shutil.py", line 250, in rmtree
    os.remove(fullname)
WindowsError: [Error 5] Access is denied: 'C:\\Miniconda\\conda-bld\\work\\.git\\objects\\pack\\pack-1f3f954ec4500439b96cf1d242b9f3fcc1829707.idx'
Command exited with code 1
~~~
for Python 2.7 Win32

and

~~~
...
Building  flac-1.3.1-0
Traceback (most recent call last):
  File "C:\Miniconda35-x64\lib\site-packages\conda_build_all\builder.py", line 200, in build
    conda_build.api.build(meta.meta, config=config)
AttributeError: module 'conda_build' has no attribute 'api'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "C:\Miniconda35-x64\Scripts\conda-build-all-script.py", line 11, in <module>
    load_entry_point('conda-build-all==1.0.0', 'console_scripts', 'conda-build-all')()
  File "C:\Miniconda35-x64\lib\site-packages\conda_build_all\cli.py", line 90, in main
    b.main()
  File "C:\Miniconda35-x64\lib\site-packages\conda_build_all\builder.py", line 263, in main
    built_dist_location = self.build(meta, build_config)
  File "C:\Miniconda35-x64\lib\site-packages\conda_build_all\builder.py", line 204, in build
    return bldpkg_path(build.build(meta.meta))
  File "C:\Miniconda35-x64\lib\site-packages\conda_build_all\build.py", line 33, in build
    shutil.rmtree(conda_build.source.WORK_DIR)
  File "C:\Miniconda35-x64\lib\shutil.py", line 488, in rmtree
    return _rmtree_unsafe(path, onerror)
  File "C:\Miniconda35-x64\lib\shutil.py", line 378, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\Miniconda35-x64\lib\shutil.py", line 378, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\Miniconda35-x64\lib\shutil.py", line 378, in _rmtree_unsafe
    _rmtree_unsafe(fullname, onerror)
  File "C:\Miniconda35-x64\lib\shutil.py", line 383, in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
  File "C:\Miniconda35-x64\lib\shutil.py", line 381, in _rmtree_unsafe
    os.unlink(fullname)
PermissionError: [WinError 5] Access is denied: 'C:\\Miniconda35-x64\\conda-bld\\work\\.git\\objects\\pack\\pack-1f3f954ec4500439b96cf1d242b9f3fcc1829707.idx'
~~~

Do you maybe have a hunch what may cause these errors (and if this is something that I can help?)

File: recipes/flac/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{% set name = "flac" %}
+{% set version = "1.3.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/xiph/flac.git
+  git_rev: {{ version }}
```
Comment (scopatz) [inline recipes/flac/meta.yaml]: Please use a tarball rather  than a repo

**Comment by scopatz** [comment]
It seems like you don't have permissions on the repo. Use a tarball instead.

File: recipes/flac/meta.yaml (hunk 1 above)
Comment (dschreij) [inline recipes/flac/meta.yaml]: Thanks, I've fixed that by point to the url of the tar.gz of that release on github. However, the problems seems to pertain to sdl2_mixer, not flac:

~~~
Building  sdl2_mixer-2.0.1-vc10_0
Traceback (most recent call last):
  File "C:\Miniconda3\lib\site-packages\conda_build_all\builder.py", line 200, in build
    conda_build.api.build(meta.meta, config=config)
AttributeError: 'module' object has no attribute 'api'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  ....
PermissionError: [WinError 5] Access is denied: 'C:\\Miniconda3\\conda-bld\\work\\.git\\objects\\pack\\pack-1f3f954ec4500439b96cf1d242b9f3fcc1829707.idx'
~~~

Another thing I noticed, is that for Python 3.5, all packages are planned to be built twice:

~~~
Resolved dependencies, will be built in the following order: 
	mad-0.15.1b-vc14_0 (will be built: True)
	mad-0.15.1b-0 (will be built: True)
	ogg-1.3.2-vc14_0 (will be built: True)
	ogg-1.3.2-0 (will be built: True)
	smpeg2-2.0.0-vc14_0 (will be built: True)
	smpeg2-2.0.0-0 (will be built: True)
	vorbis-1.3.5-vc14_0 (will be built: True)
	vorbis-1.3.5-0 (will be built: True)
	flac-1.3.1-vc14_0 (will be built: True)
	flac-1.3.1-0 (will be built: True)
	sdl2_mixer-2.0.1-vc14_0 (will be built: True)
	sdl2_mixer-2.0.1-0 (will be built: True)
~~~

while this isn't happening for Python 3.4 or 2.7...

**Comment by jjhelmus** [comment]
@dschreij It may be easier if you add these dependencies individually as separate pull requests.  It may take longer to get them all merged but it would make debugging and review these much easier.

**Comment by dschreij** [comment]
@jjhelmus, ok, I thought I made it easier like this for you guys ;), but I will split them up in loads of small PRs then (prepare for a flood).

**Comment by dschreij** [comment]
@jjhelmus Is there a general rule about how many dependencies can be included in a PR, or is it a one package per PR policy? In other words, do I have to split up [glew and sdl2_gfx](https://github.com/conda-forge/staged-recipes/pull/1945) and [sdl2_image and webp](https://github.com/conda-forge/staged-recipes/pull/1941) too?

**Comment by jjhelmus** [comment]
One recipe per PR tends to make reviewing easier, although there is no hard and fast rule.

**Comment by dschreij** [comment]
Closing and reopening new PR

**Pull request closed without merging**